File: python/single_alarm.py
import alarm
from alarm import Alarm
from interfaces import Alarm_Behavior


class Single_Alarm(Alarm_Behavior):
    def __init__(self):
        self._alarm = None

    def add_alarm(self, alarm):
        self._alarm = alarm

    def get_alarms(self, from_time=None, to_time=None):
        result = []
        if self._alarm:
            result.append(self._alarm)
        return result

    def remove_alarm(self, alarm):
        if self._alarm and self._alarm.alarm_name == alarm.alarm_name:
            self._alarm = None

    def find_alarm_by_name(self, alarm_name):
        if self._alarm and self._alarm.alarm_name == alarm_name:
            return self._alarm
        return None

File: java/src/com/infoback/SingleAlarm.java
package com.infoback;

import java.time.LocalDateTime;
import java.util.ArrayList;

public class SingleAlarm implements IAlarmStrategy {
  private Alarm _alarm;

  @Override
  public void addAlarm(Alarm alarm) {
    this._alarm = alarm;
  }

  @Override
  public Iterable<Alarm> getAlarms(LocalDateTime fromTime, LocalDateTime toTime) {
    ArrayList<Alarm> result = new ArrayList<>();
    if (_alarm != null)
      result.add(_alarm);
    return result;
  }

  @Override
  public void removeAlarm(Alarm alarm) {
    if (_alarm != null && _alarm.getAlarmName().equals(alarm.getAlarmName()))
      _alarm = null;
  }

  @Override
  public Alarm findAlarmByName(String alarmName) {
    if (_alarm != null && _alarm.getAlarmName().equals(alarmName))
      return _alarm;
    return null;
  }
}
